contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1357,6 +1361,60 @@ select_common_type_for_isnull(ParseState *pstate, List *exprs)
 	return ptype;
 }
 
+static Oid
+select_common_type_for_coalesce_function(ParseState *pstate, List *exprs)
+{
+	Node	   *pexpr;
+	Oid		   ptype;
+	ListCell   *lc;
+	Oid		   commontype = InvalidOid;
+	int curr_precedence;
+
+	Assert(exprs != NIL);
+
+	if (exprs->length < 2)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("COALESCE function should have at least 2 arguments")));
+
+	foreach(lc, exprs)
+	{
+		pexpr = (Node *) lfirst(lc);
+		ptype = exprType(pexpr);
+
+		/* Check if arg is NULL literal */
+		if (IsA(pexpr, Const) && ((Const *) pexpr)->constisnull && ptype == UNKNOWNOID)
+			continue;
+
+		/* If the arg is non-null string literal */
+		if (ptype == UNKNOWNOID)
+		{
+			Oid curr_oid = get_sys_varcharoid();
+			if (commontype == InvalidOid 
+				|| tsql_get_type_precedence(curr_oid) < curr_precedence)
+			{
+				commontype = curr_oid;
+				curr_precedence = tsql_get_type_precedence(curr_oid);
```

Comment:
Define `tsql_get_type_precedence(curr_oid)` and `tsql_get_type_precedence(ptype)` once and then reuse their values

Reply:
done